build/toolchain/apple/toolchain.gni

```diff
@@ -89,6 +129,34 @@ template("apple_toolchain") {
     } else {
       toolchain_cc_wrapper = cc_wrapper
     }
+    assert(!(toolchain_uses_remoteexec && toolchain_uses_goma),
+           "Goma and re-client can't be used together.")
+    assert(!(toolchain_cc_wrapper != "" && toolchain_uses_remoteexec),
+           "re-client and cc_wrapper can't be used together.")
+    assert(!(toolchain_cc_wrapper != "" && toolchain_uses_goma),
+           "Goma and cc_wrapper can't be used together.")
+
+    if (defined(toolchain_args.use_lld)) {
+      toolchain_uses_lld = toolchain_args.use_lld
+    } else {
+      toolchain_uses_lld = use_lld
+    }
+
+    # The value of all global variables (such as `is_component_build`) is the
+    # one from the default toolchain when evaluating a secondary toolchain
+    # (see https://crbug.com/gn/286). This mean that the value may change when
+    # evaluating target/configs in the new toolchain if the variable default
+    # value depends on variable set in `toolchain_args`.
+    #
+    # For this reason, "ios" needs to override `is_component_build` as its
+    # default value depends on `current_os`. Use the overridden value if it
+    # is set in `toolchain_args`.
+    if (defined(toolchain_args.is_component_build)) {
+      toolchain_is_component_build = toolchain_args.is_component_build
+    } else {
+      toolchain_is_component_build = is_component_build
+    }
+
```

Comment:
Lines 160–173 aren't in the source revision:
https://chromium.googlesource.com/chromium/src/+/114.0.5735.331/build/toolchain/apple/toolchain.gni#160